include/api.php

```diff
@@ -90,11 +91,15 @@ function api_source()
 	}
 
 	// Support for known clients that doesn't send a source name
-	if (strpos($_SERVER['HTTP_USER_AGENT'], "Twidere") !== false) {
-		return "Twidere";
+	if (!empty($_SERVER['HTTP_USER_AGENT'])) {
+		if(strpos($_SERVER['HTTP_USER_AGENT'], "Twidere") !== false) {
+			return "Twidere";
+		}
+
+		logger("Unrecognized user-agent ".$_SERVER['HTTP_USER_AGENT'], LOGGER_DEBUG);
 	}
 
-	logger("Unrecognized user-agent ".$_SERVER['HTTP_USER_AGENT'], LOGGER_DEBUG);
+	logger("Empty user-agent", LOGGER_DEBUG);
```

Comment:
Shouldn't this be better part of some ```else``` construct?

Reply:
Indeed, good catch.